### PR TITLE
NEXUS-10659-iq-server-upgrade

### PR DESCRIPTION
**Pull request by dulanism**
JIRA: https://issues.sonatype.org/browse/NEXUS-10659

Notes:
- Cherry pick these changes into @jwayman's branch
- Added 'Health Check' to the list of Repository content that can be upgraded (see 'Upgrading Content')

- Added 'Crowd' config as s 'Content' option

### CODE REVIEW

**Comment by dulanism** [comment]
Thinking, retrospectively, that I should add a note that the upgrade of this feature will not work if the user only has an NXRM3 trial license. Thoughts?

**Comment by joedragons** [comment]
@dulanism Are there NX3 trial licenses yet?  I thought that wasn't done.  Maybe I'm misremembering.
Also (without building) looks like the PR does not add to the list of items that you can migrate further down in the guide.  Just adds detail about the area.  I can quote better once I build if this is unclear.

**Comment by dulanism** [comment]
@joedragons I'm the mis-rememberer, as I now looking back as some notes. Thanks for the heads up, about the content list. I'll update in the next commit.

File: chapter-upgrading.asciidoc [hunk 1]
```diff
@@ -90,6 +90,21 @@ the file is, and they obtain it from the file system, directly.
 [[upgrade-content-prep]]
 === Preparing Content for Upgrade
 
+*IQ Server*
+
+{inrmonly}
+
+If upgrading your {iq} integration, ensure that your licenses for both versions 2 and 3 have the capability 
+enabled. Your configuration for 'IQ Server URL', 'Username', 'Password', and 'Request Timeout' will be included
```
Comment (joedragons) [inline chapter-upgrading.asciidoc]: +0 to first sentence as I cannot remember if this is true.  I am pretty sure other config just migrates over and no pre-treatment is needed.  Looks like @bentmann tested this maybe he remembers.  I can check if you like, let me know.  If you've experienced this then you can ignore this comment=)

File: chapter-upgrading.asciidoc [hunk 2]
```diff
@@ -90,6 +90,21 @@ the file is, and they obtain it from the file system, directly.
 [[upgrade-content-prep]]
 === Preparing Content for Upgrade
 
+*IQ Server*
+
+{inrmonly}
+
+If upgrading your {iq} integration, ensure that your licenses for both versions 2 and 3 have the capability 
+enabled. Your configuration for 'IQ Server URL', 'Username', 'Password', and 'Request Timeout' will be included 
+in the upgrade. Additionally, scanner properties, trust store usage, and enabled {iq} connection can be
```
Comment (joedragons) [inline chapter-upgrading.asciidoc]: You say "can" here is it optional?  I don't recall an option.  Alternatively, you might be saying/inferring these fields may be blank and thus arn't technically migrated as blank.  If that's so, might be worth a rephrase to avoid the misinterpretation stated above.

File: chapter-upgrading.asciidoc [hunk 3]
```diff
@@ -90,6 +90,21 @@ the file is, and they obtain it from the file system, directly.
...
 === Preparing Content for Upgrade
 
+*IQ Server*
+
+{inrmonly}
+
+If upgrading your {iq} integration, ensure that your licenses for both versions 2 and 3 have the capability 
+enabled. Your configuration for 'IQ Server URL', 'Username', 'Password', and 'Request Timeout' will be included 
+in the upgrade. Additionally, scanner properties, trust store usage, and enabled {iq} connection can be 
+replicated from versions 2 to 3. As as best practice, be sure to encrypt your {iq} password before initiating
```
Comment (joedragons) [inline chapter-upgrading.asciidoc]: Should be encrypted by us already, I think.  Maybe dev (or you) know more than me.

Comment (bentmann) [inline chapter-upgrading.asciidoc]: What Joe said. Really not sure what message we try to get across, pretty much any piece of Nexus UI that I remember expects end users to enter credentials as is, i.e. in clear, it's up to the code to encrypt/obfuscate those passwords when persisting them or as here when transmitting them.

File: chapter-upgrading.asciidoc (hunk 1 above)
Comment (bentmann) [inline chapter-upgrading.asciidoc]: One cannot even select the IQ config for upgrade if the license in NX3 doesn't enable it (cf. https://github.com/sonatype/nexus-internal/commit/ebf37c20517d1211e6480c8b324e60ab7804ba40#diff-274d35362e4e7f6e304ff7af4e518668R77). If the license in NX2 doesn't enable IQ it will just look like IQ isn't configured (cf. https://github.com/sonatype/nexus2-internal/commit/acf91a42695c92f203e26f8ac1d9300d5c1dee38#diff-2a8caff70f970cf2a042c36d004b91a1R75)

Comment (joedragons) [inline chapter-upgrading.asciidoc]: Ok, cool, then I think this sentence is fine.  Thanks @bentmann .

Comment (dulanism) [inline chapter-upgrading.asciidoc]: Ah, good to know. I think that should be noted. In the process of testing. Thanks.

Comment (bentmann) [inline chapter-upgrading.asciidoc]: What I described is the implementation, I don't see why an end user would care about that level of detail. I find your current sentence rather sufficient.

File: chapter-upgrading.asciidoc [hunk 4]
```diff
@@ -90,6 +90,21 @@ the file is, and they obtain it from the file system, directly.
 [[upgrade-content-prep]]
 === Preparing Content for Upgrade
 
+*IQ Server*
+
+{inrmonly}
+
+If upgrading your {iq} integration, ensure that your licenses for both versions 2 and 3 have the capability
```
Comment (bentmann) [inline chapter-upgrading.asciidoc]: "capability" already has a certain and different meaning in NX. For licenses, we typically speak of features AFAIK.

File: chapter-upgrading.asciidoc (hunk 2 above)
Comment (dulanism) [inline chapter-upgrading.asciidoc]: Consider this update: 

> Additionally, if configured, scanner properties, trust store usage, and the enabled {iq} 
connection itself will be replicated from versions 2 to 3.

File: chapter-upgrading.asciidoc [hunk 5]
```diff
@@ -90,6 +99,20 @@ the file is, and they obtain it from the file system, directly.
 [[upgrade-content-prep]]
 === Preparing Content for Upgrade
 
+*IQ Server*
+
+{inrmonly}
+
+If upgrading {iq}, ensure that your licenses include the integration for both versions 2 and 3. Your
```
Comment (bentmann) [inline chapter-upgrading.asciidoc]: "If upgrading Nexus IQ Server", we might want to add "settings/configuration" or the like to make it clear that this is about the configuration in NX, not the IQ Server itself.

File: chapter-upgrading.asciidoc [hunk 6]
```diff
@@ -90,6 +99,20 @@ the file is, and they obtain it from the file system, directly.
 [[upgrade-content-prep]]
 === Preparing Content for Upgrade
 
+*IQ Server*
+
+{inrmonly}
+
+If upgrading {iq}, ensure that your licenses include the integration for both versions 2 and 3. Your 
+configuration for 'IQ Server URL', 'Username', 'Password', and 'Request Timeout' will be included in the upgrade. 
+Additionally, if configured, scanner properties, trust store usage, and the enabled {iq} connection itself will
```
Comment (bentmann) [inline chapter-upgrading.asciidoc]: Last I heard, "scan" or "scanner" are frowned upon to describe our tooling to the public. Potential alternatives could be "analysis properties" or "custom configuration properties".

**Comment by bentmann** [comment]
+1

File: chapter-upgrading.asciidoc (hunk 2 above)
Comment (joedragons) [inline chapter-upgrading.asciidoc]: Looks good.

File: chapter-upgrading.asciidoc [hunk 7]
```diff
@@ -90,6 +99,20 @@ the file is, and they obtain it from the file system, directly.
 [[upgrade-content-prep]]
 === Preparing Content for Upgrade
 
+*IQ Server*
+
+{inrmonly}
+
+If upgrading {iq} settings and configuration, ensure that your licenses include the integration for both versions 
+2 and 3. Your configuration for 'IQ Server URL', 'Username', 'Password', and 'Request Timeout' will be included 
+in the upgrade. Additionally, if configured, analysis properties, trust store usage, and the enabled {iq}
```
Comment (joedragons) [inline chapter-upgrading.asciidoc]: Minor (comma drama): I'm not sure about "proper usage" but visually, removing the comma before if I think makes the conditionals look less like a part of the immediately following comma deliniated list.

File: chapter-upgrading.asciidoc [hunk 8]
```diff
@@ -265,17 +284,13 @@ base URL (including context path) of your version 2 server (e.g. +http://localho
...
 
 Content:: This screen displays checkboxes for compatible component formats ('Repositories'), security features 
-('Security'), and server configuration ('System'). For 'Repositories' you can select 'User-Managed Repositories'  
-and 'Repository Targets', the latter of which are regular expressions that upgrade to JEXL format, used for 
-content selectors. For 'Security' you can choose from 'Anonymous', 'LDAP Configuration', 'NuGet API-Key', 
-'Realms', 'Roles', 'SSL Certificates', 'Target Privileges', 'Users', and 'User Tokens'. For 'System' you can 
-select 'Email' and 'HTTP Configuration' contents.
+('Security'), and server configuration ('System'). For 'Repositories' you can select 'User-Managed Repositories', 
+'Repository Targets', and 'Health Check'. For 'Security' you can choose from 'Anonymous', 'LDAP Configuration', 
+'NuGet API-Key', 'Realms', 'Roles', 'SSL Certificates', 'Target Privileges', 'Users', and 'User Tokens'. For
```
Comment (joedragons) [inline chapter-upgrading.asciidoc]: Crowd is missing here, but I believe there's a distinct ticket for that.  If you're not going to do more than add to list, this could be easily done here saving you another PR (tho maybe that's taboo=)).

**Comment by joedragons** [comment]
+1.  It is interesting to note that your list of content does not 1-1 match the app because yours is alphabetical and the app is not.  If anything, I think that's a fault of the app tho.
